spec/concierge_spec.rb

```diff
@@ -0,0 +1,170 @@
+require_relative 'spec_helper'
+
+describe 'Lodging::Concierge class' do
+  #reset CSV data
+  before(:each) do
+    CSV.open('data/all_hotel_rooms.csv', 'w', headers: true) do |csv|
+      csv << ['room_number','cost','status', 'reserved_dates']
+    end
+
+    CSV.open('data/reservations.csv', 'w', headers: true) do |csv|
+      csv << ['room_number','cost','status','check_in','check_out']
+    end
+  end
+
+  let(:concierge){
+    Lodging::Concierge.new(10)
+  }
+
+  describe 'all rooms method' do
+    it 'can view data in all_hotel_rooms csv file' do
+
+      expect(concierge.all_rooms).must_be_instance_of Array
+      expect(concierge.all_rooms.any? Hash).must_equal true
+      expect(concierge.all_rooms.length).must_equal 10
+
+    end
+  end
+
+  describe 'new reservation method' do
+
+    it 'creates new reservations' do
+      concierge.all_rooms
+
+      concierge.new_reservation('Sept 9, 2018', 'Sept 12, 2018')
+
+      first = concierge.all_rooms[0][:reserved_dates].first
+      last = concierge.all_rooms[0][:reserved_dates].last
+      status = concierge.all_rooms[0][:status]
+
+      expect(status).must_equal 'unavailable'
+      expect(first).must_equal Date.parse('Sept 9, 2018')
+      expect(last).must_equal Date.parse('Sept 12, 2018')
+
+    end
+
+    it 'errors if invalid date is entered' do
+      expect{concierge.new_reservation('August 7, 2019', 'August 28, 2018')}.must_raise ArgumentError
+    end
+
+    it 'can reserve room on last day of previous res' do
+      tiny_hotel = Lodging::Concierge.new(1)
+
+
+      tiny_hotel.new_reservation('Sept 9, 2018', 'Sept 12, 2018')
+      tiny_hotel.new_reservation('Sept 12, 2018', 'Sept 14, 2018')
+      tiny_hotel.all_rooms
+
+      expect(tiny_hotel.all_rooms[0][:reserved_dates].length).must_equal 7
+
+    end
```

Comment:
What about testing if you can do multiple reservations.

What happens if you reserve rooms and no room is left available.

These things should be tested!